### ISSUE TITLE
NPCs fail to replace empty rods

### ISSUE THREAD

**Issue by LePiromano** [Regalis11/Barotrauma#4106]
- [yes] I have searched the issue tracker to check if the issue has already been reported.

**Description**
It's been here for a while, i think could be fixed easily, probably is a problem between the AI and the "Clean Up" interaction. I'm still not 100% sure of how it happens, but what i wrote in "steps to reproduce" is what i know. NPCs also start spamming "I cant find the item i need" sometimes when it happens.
![20201019173951_1](https://user-images.githubusercontent.com/55637865/96509408-5a71bc00-1232-11eb-8d3b-a9c45912b807.jpg)

**Steps To Reproduce**
Make the rod inside the reactor empty, all lockers where fuel rods should be (probably related to the locker's tag) have to be full of items, give any npc the order to power on the reactor. If it worked the npc should stand next to the reactor, sometimes they will the empty rod, sometimes they wont.

**Version**
Unstable V0.10.6.2

**Additional information**
probably caused by the cleanup interaction, maybe they prioritize not leaving items in the floor instead of giving priority to the reactor.

**Comment by NyarlathotepZERO** [comment]
I wish npc's would take it one step further and drop the spent rod into a deconstructor

**Comment by Xeavas** [comment]
To elaborate after some testing.

just needs to have 2 criteria for bug to happen
1) reactor tagged cabinet is full
2) empty rod in the inventory of bot tasked to man reactor
switching to them and dropping the empty rod results them going back to functioning normally (taking fresh rod out of full locker, and loading into reactor)

My conclusion : There's some error in trying to put back the rods into tagged cabinets instead of any nearby spare containers or just simply dropping them on the floor.

![aireccab](https://user-images.githubusercontent.com/72674994/97912335-a3278a00-1d87-11eb-9e61-1500d1a7da80.jpg)


This is also observed with coilgun ammunition boxes and shelves.

**Comment by FLundin** [comment]
I have noticed that this happens when the engineer have empty fuel rods in their inventory. it you stun them with the stunbaton and take out the empty rods from their inventory, they can load the fuel in the reactor

**Comment by itchyOwl** [comment]
Fixed in Regalis11/Barotrauma-development@e5e2f69.

I'm pretty sure that tickets #3808 #4106 #4044 are all about the same issue: the bots failed to make the difference between empty and non-empty items. This was the case with fuel rods and I'm pretty sure that it's the same thing with the coilgun ammo too. It's basically the same code, so fixing one should fix the other.

Steps to repro:
1) Make the fuel rods/ammo boxes empty (easy in the sub editor)
2) Insert the empty item in the reactor/turret loader
3) Order a bot to operate the item
4) See how the bot gets stuck in the "Get Item" state. In a debug build the console floods "Removing SUBobjective get item of contain item, because it is completed"

So basically a part of the code told the bot to use only items that have some juice in them, but another part of the code the bots were told that empty items are good enough (just because the default value was 0 and not 1).

**Comment by Xeavas** [comment]
[0.1100.0.5] - Unable to replicate 👍

**Comment by Rokvach** [comment]
Closing.
